src/lib/rng/jitter_rng/jitter_rng.h

```diff
@@ -0,0 +1,40 @@
+/*
+* CPU Jitter Random Number Generator
+* (C) 2024 Planck Security S.A.
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_JITTER_RNG_H_
+#define BOTAN_JITTER_RNG_H_
+
+#include <botan/rng.h>
+
+namespace Botan {
+
+/*
+* RNG using libjitterentropy.
+*/
+class BOTAN_PUBLIC_API(3, 6) Jitter_RNG final : public RandomNumberGenerator {
+   public:
+      Jitter_RNG();
+      ~Jitter_RNG();
+
+      std::string name() const override;
+
+      bool is_seeded() const override;
+
+      bool accepts_input() const override;
+
+      void clear() override;
+
+   protected:
+      void fill_bytes_with_input(std::span<uint8_t> out, std::span<const uint8_t> in) override;
+
+   private:
+      struct Rand_Data;
```

Comment:
I have my doubts about this. I think the two `struct Jitter_RNG_Internal` declarations would be declaring different structures, since they are in different namespaces.
Actually, while writing this, I made a minimal example of this, the compiler complains about an ambiguous reference to the structure and mentions both declarations (in `library namespace` and in `library namespace::anonymous namespace`).

Reply:
But, structure declarations don't actually generate symbols, the symbols are generated instead for the functions and members of the structure (the parts that you could meaningfully refer to using pointers in code).
I presume Botan uses a compiler flag to make the default visibility hidden, because you have `BOTAN_DLL` that serves the purpose of exposing symbols that would otherwise be hidden. Perhaps then, there is no need to use an anonymous namespace.
Moving the declaration outside of the interface class (`class BOTAN_DLL Jitter_RNG`) might however make sense because visibility attributes propagate to nested definitions, as can be seen [in this SO question](https://stackoverflow.com/questions/1247745/default-visibility-of-c-class-struct-members). But maybe a better solution that also avoids polluting the `Botan` namespace would be to use an attribute to manually prevent the exposition of symbols within the internal structure declaration. It could look like this:
```
#define BOTAN_INTERNAL __attribute__((visibility("hidden")))
...........................................................
class BOTAN_DLL Jitter_RNG
{
   class BOTAN_INTERNAL Rand_Data;
};
```
I don't know if this approach is currently being used in the library or if it works at all (I will test it and post about it in a moment).

Reply:
furthermore, if in the `.cpp` file, all member functions of the implementation class (`Rand_Data`) are inline (either by defining them inside the class definition or explicitly declaring them to be `inline`), they won't generate any symbols.

Reply:
I tested the following approach:
```
#define BOTAN_INTERNAL [[gnu::visibility("hidden")]]
...........................................................
class BOTAN_DLL Jitter_RNG
{
   ...
   class BOTAN_INTERNAL Rand_Data;
};
```
It effectively hides symbols inside `Rand_Data`
The following also works:
```
// in header
class BOTAN_DLL Jitter_RNG
{
   ...
   class Rand_Data;
};

// in .cpp
class BOTAN_INTERNAL Jitter_RNG::Rand_Data
{
   ...
};
```

Reply:
... I see the problem now. In windows there's no way to "use an attribute to manually prevent the exposition of symbols". There's no undoing the `__declspec(dllexport)`

Reply:
Looks like putting both definition and forward declaration into an anonymous namespace under Botan works, at least on macOS clang:

https://github.com/plancksecurity/botan-jitter/blob/jitter_review_2024-10-02/src/lib/rng/jitter_rng/jitter_rng.h#L13
https://github.com/plancksecurity/botan-jitter/blob/jitter_review_2024-10-02/src/lib/rng/jitter_rng/jitter_rng.cpp#L12